exporter/helpers.go

```diff
@@ -247,12 +247,15 @@ func getDIMMEndpoints(url, host string, client *retryablehttp.Client) (oem.Colle
 	if err != nil {
 		return dimms, err
 	}
-	defer resp.Body.Close()
+	defer common.EmptyAndCloseBody(resp)
 	if !(resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices) {
 		if resp.StatusCode == http.StatusNotFound {
 			for retryCount < 1 && resp.StatusCode == http.StatusNotFound {
 				time.Sleep(client.RetryWaitMin)
 				resp, err = common.DoRequest(client, req)
+				if resp != nil {
```

Comment:
Are you sure didn't want to do if err == nil ? (yes I realize this is similar considering the DoRequest wrapper)